### PR TITLE
feat: add schema option UseOneofFields

### PR DESCRIPTION
**Pull request by alfredgunnar**
When set, if the message contains any oneof fields, a BigQuery STRING field will be created
for each oneof group. It will then contain the name of the oneof that is set. This gives
support for having oneof groups where one of the fields in the group is an empty message.

This commit also fixes the behavior of InferSchema to skip fields that are empty messages
since that is not supported by BigQuery.

### CODE REVIEW

File: encoding/protobq/marshal.go [hunk 1]
```diff
@@ -66,13 +66,23 @@ func (o MarshalOptions) marshalMessage(msg protoreflect.Message) (map[string]big
...
 		}
 		return true
 	})
 	if returnErr != nil {
 		return nil, returnErr
 	}
+	if o.Schema.UseOneofFields {
+		for i := 0; i < msg.Descriptor().Oneofs().Len(); i++ {
+			oneof := msg.Descriptor().Oneofs().Get(i)
+			result[string(oneof.Name())] = string(msg.WhichOneof(oneof).Name())
```
Comment (odsod) [inline encoding/protobq/marshal.go:83]: What if the oneof is not set here? What will `Name()` return?

File: encoding/protobq/marshal_test.go [hunk 2]
```diff
@@ -266,6 +266,68 @@ func TestMarshalOptions_Marshal(t *testing.T) {
...
+			expected: map[string]bigquery.Value{
+				"oneof_fields_1": "oneof_empty_message_1",
+				"oneof_fields_2": "oneof_message",
+				"oneof_message": map[string]bigquery.Value{
+					"string_value": "value",
+				},
+			},
+		},
+
+		{
```
Comment (odsod) [inline encoding/protobq/marshal_test.go:309]: can we add a test case with a completely empty message?

File: encoding/protobq/marshal.go (hunk 1 above)
Comment (alfredgunnar) [reply encoding/protobq/marshal.go:83]: Oooh! Nice catch. That's a nil pointer. 😁

File: encoding/protobq/marshal_test.go (hunk 2 above)
Comment (alfredgunnar) [reply encoding/protobq/marshal_test.go:309]: YES! That's definitely needed.

**Comment by alfredgunnar** [comment]
@odsod Thanks for the feedback! Very nice bug catch you did there. Added a test case for both marshal and unmarshal, but it only required changes for marshal.

File: encoding/protobq/marshal.go (hunk 1 above)
Comment (odsod) [reply encoding/protobq/marshal.go]: :smile_cat: I felt a disturbance in the force

**Review by odsod — APPROVED** [review]
Awesome stuff!
